### PR TITLE
JavaScript Capstone Project Final

### PR DESCRIPTION
**Pull request by KUBAHO3**
In this JavaScript Capstone Project, We have:

- Use JavaScript to make websites dynamic and build basic single page apps.
- Use ES6 syntax.
- Use ES6 modules.
- Use callbacks and promises.
- Use webpack.
- Apply JavaScript best practices and language style guides in code.
- Use AAA pattern for unit tests.
- Write units tests for a JavaScript app.
- Follow Gitflow.
- Solve simple git conflicts.
- Send and receive data from an API.
- Use API documentation.
- Understand and use JSON.
- Make JavaScript code asynchronous.
- Perform a code review for a team member.

To run this project in your system please consider the following practices:

- [ ] clone this GitHub repository `git clone https://github.com/KUBAHO3/JS-Capstone` and navigate to the feature branch `cd JS-Capstone`, `git checkout feature`
- [ ] install all node packages by `npm install`
- [ ] run the following command in your terminal to build the system on your dev `npm run build` or `npm start`
- [ ] If you used `npm run build` then navigate to the `base_url/dist/`

### CODE REVIEW

File: src/module/_tests_/countComments.test.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+test('count items', () => {
+  document.body.innerHTML = `<ul class="comments-lists">
+                                <p>12/03/2021 Lawrence: I love meat</p>
+                                <p>12/05/2023 Kenny: Cool show</p>
+                                <p>12/03/2021 Heaven: Dessert are the best</p>
+                                <p>1/08/2022 Sammy: Nothing beats chicken</p>
+                            </ul>`;
+  const myListOfComments = document.querySelector('.comments-lists');
+  const nbOfComments = countComments(myListOfComments);
+  expect(nbOfComments).toBe(4);
```
Comment (codepantha) [inline src/module/_tests_/countComments.test.js:12]: - [ ] Please also consider the edge case where there are 0 comments on the page. Your tests should cover that scenario 👍

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi team 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Good commit history
✔️ No linter errors

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/module/_tests_/countComments.test.js (hunk 1 above)
Comment (KUBAHO3) [reply src/module/_tests_/countComments.test.js:12]: Thank you for the review! we have added it.

**Review by codepantha — APPROVED** [review]
# STATUS: APPROVED 🟢 ✔️ 

## Hi team 👋 

Great work on making the requested changes 👏 
Your project is now complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
